core/capabilities/remote/executable/request/server_request.go

```diff
@@ -164,6 +164,22 @@ func NewServerRequest(capability capabilities.ExecutableCapability, method strin
 }
 
 func (e *ServerRequest) OnMessage(ctx context.Context, msg *types.MessageBody) error {
+	// Create a new span for this message processing
+	ctx, span := beholder.GetTracer().Start(ctx, "server_request.process_message",
+		trace.WithAttributes(
+			attribute.String("p2p_id", e.capabilityPeerID.String()),
+			attribute.String("trace_id", msg.TraceId),
```

Comment:
Is this actually linking the trace? Doesn't seem right...🤔

Reply:
Another one: https://github.com/grpc/grpc-go/blob/master/stats/opentelemetry/grpc_trace_bin_propagator.go#L110-L118
I wonder if span ID is important to propagate too :thinking:

Reply:
And it looks like trace flags are only a byte so should consider that too :shrug: